pom.xml

```
@@ -213,4 +196,70 @@
             </plugin>
         </plugins>
     </build>
-</project>
\ No newline at end of file
+    <profiles>
+    <profile>
+        <id>spark24</id>
+        <properties>
+            <scala.binary.version>2.11</scala.binary.version>
+            <scala.version>2.11.12</scala.version>
+            <spark.version>2.4.6</spark.version>
+        </properties>
+        <dependencies>
+            <dependency>
+                <groupId>org.scalatest</groupId>
+                <artifactId>scalatest_${scala.binary.version}</artifactId>
+                <version>3.0.5</version>
+                <scope>test</scope>
+            </dependency>
+            <dependency>
+                <groupId>com.microsoft.sqlserver</groupId>
+                <artifactId>mssql-jdbc</artifactId>
+                <version>7.2.1.jre8</version>
+            </dependency>
+        </dependencies>
+    </profile>
+    <profile>
+        <id>spark30</id>
+        <properties>
+            <scala.binary.version>2.12</scala.binary.version>
+            <scala.version>2.12.11</scala.version>
+            <spark.version>3.0.0</spark.version>
+        </properties>
+        <dependencies>
+        <dependency>
+            <groupId>org.scalatest</groupId>
+            <artifactId>scalatest_${scala.binary.version}</artifactId>
+            <version>3.0.8</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.microsoft.sqlserver</groupId>
+            <artifactId>mssql-jdbc</artifactId>
+            <version>8.2.2.jre8</version>
+        </dependency>
+        </dependencies>
+    </profile>
+        <profile>
+            <id>spark311</id>
+            <properties>
+                <scala.binary.version>2.12</scala.binary.version>
+                <scala.version>2.12.10</scala.version>
+                <spark.version>3.1.1</spark.version>
+            </properties>
+            <dependencies>
+                <dependency>
+                    <groupId>org.scalatest</groupId>
+                    <artifactId>scalatest_${scala.binary.version}</artifactId>
+                    <version>3.2.6</version>
+                    <scope>test</scope>
+                </dependency>
+                <dependency>
+                    <groupId>com.microsoft.sqlserver</groupId>
+                    <artifactId>mssql-jdbc</artifactId>
+                    <version>8.2.1.jre8</version>
```

Why downgrade the version for the 3.1 connector?
```suggestion
                    <version>8.2.2.jre8</version>
```